CodeJam.Main/IO/StreamExtensions.cs

```diff
@@ -110,6 +111,32 @@ public static string ReadAsString(this Stream stream, Encoding? encoding = null)
 				return await reader.ReadToEndAsync().ConfigureAwait(false);
 		}
 
+		/// <summary>
+		/// Returns content of the stream as a string.
+		/// </summary>
+		/// <param name="stream">The stream to read.</param>
+		/// <param name="encoding">The character encoding to use.</param>
+		/// <param name="cancellationToken">Token of cancellation of the operation.</param>
+		public static async Task<string> ReadAsStringAsync(
+			this Stream stream,
+			Encoding? encoding = null,
+			CancellationToken cancellationToken = default)
+		{
+			Code.NotNull(stream, nameof(stream));
+			using var register = cancellationToken.Register(() => stream.Close());
+
+			try
+			{
+				using var reader = stream.ToStreamReader(encoding, true);
+				return await reader.ReadToEndAsync().ConfigureAwait(false);
+			}
+			catch
+			{
+				if (!cancellationToken.IsCancellationRequested) throw;
```

Comment:
Hi!

I'm not sure but it looks like there is something wrong with the check. The code will throw `OperationCanceledException` if `cancellationToken.IsCancellationRequested` is false. It will be very misleading, as for me.